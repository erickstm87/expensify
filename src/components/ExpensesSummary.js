import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    console.log('i am being read');
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal/100).format('$0.0.00')
    return (
        // <div className="page-header">
        //     <div className="content-container">
        <div>
                <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
            </div>
        // </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses),
        statement: () => {
            console.log('being read');
        }
    };
};

export default connect(mapStateToProps)(ExpensesSummary);