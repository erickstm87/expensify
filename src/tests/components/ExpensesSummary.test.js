import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={234} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={13} expensesTotal={234323463456} />);
    expect(wrapper).toMatchSnapshot();
});