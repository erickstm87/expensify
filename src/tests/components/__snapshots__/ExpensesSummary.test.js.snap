// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.3
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    13
     
    expenses
     totalling 
    $2343234634.6
  </h1>
</div>
`;
